src/ddmd/expressionsem.d

```diff
@@ -9009,16 +9012,16 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         // Special handling for array comparisons
         if (!(t1.ty == Tarray && t2.ty == Tarray && needsDirectEq(t1, t2)))
         {
-            if (!arrayTypeCompatible(exp.loc, exp.e1.type, exp.e2.type))
```

Comment:
Looks as though the formatting got mucked up at this point.

Reply:
Indeed, also these lines shouldn't be touched.